packages/unmock/src/index.ts

```diff
@@ -0,0 +1,80 @@
+import FailingBackend from "./backend/failing-backend";
+import FailingLogger from "./logger/failing-logger";
+import { IUnmockInternalOptions, IUnmockOptions } from "./options";
+import FailingPersistence from "./persistence/failing-persistence";
+import getToken from "./token";
+import {
+  buildPath,
+  endReporter,
+  hostIsWhitelisted,
+  UNMOCK_UA_HEADER_NAME,
+} from "./util";
+
+// top-level exports
+export { IUnmockInternalOptions, IUnmockOptions } from "./options";
+export { IBackend } from "./backend";
+export { ILogger } from "./logger";
+export { IPersistence } from "./persistence";
+export { IPersistableData } from "./util";
+export const util = {
+  UNMOCK_UA_HEADER_NAME,
+  buildPath,
+  endReporter,
+  hostIsWhitelisted,
+};
+
+export const defaultOptions: IUnmockInternalOptions = {
+  backend: new FailingBackend(),
```

Comment:
Is the `FailingX` the recommended way of doing things?
I had a ticket open to refactor the whole `UnmockOptions` here as well. I thought it should be a class (instead of an interface), and then we can define the default in the constructor itself. That would eliminate the need for these `FailingX` classes.
(It would also then simplify e.g. `endReporter`, etc, where most of the parameters are taken from the `UnmockOptions` object).

Reply:
interesting idea!
the failing thing is a convention i've taken from the java world, but i don't claim it's the best way to do it!